pkg/templating/reconciler.go

```diff
@@ -198,6 +199,27 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.kube.Status().Update(ctx, cr), errUpdateResourceStatus)
 	}
 
+	if meta.WasDeleted(cr) {
+		deleting, err := r.children.Delete(ctx, childResources)
```

Comment:
Is there any value in confirming that we're the controller reference of the children we're about to delete before we do so, now that we're doing this manually?

Reply:
It's a rare case but it'd be consistent if we do that both for deletion as well as during `Apply`. I'll add it.